DataDog/datadog-agent
# Create  `forwarder/internal/retry` package.

### What does this PR do?

This PR moves the logic related to the retry queue from the `forwarder` package into a new sub-package `forwarder/internal/retry`.
In order to do so, the `transaction` type was moved to its own sub-package `forwarder/transaction`.

The name of the feature evolved over the time which created inconstancy in the naming of the types. This PR also improves the naming of several types.

**No code change is expected in this PR**

### Motivation

`forwarder` package starts to be too big.
AC-811.

### Additional Notes

**Reviewing commit by commit is strongly recommended.**

### Describe your test plan

* Check goexpvars and the telemetry is the same as before (Checking the variable names is enough). 
* Check the forwarder can store transaction on disk and read them.
